sdk/iot/Azure.Iot.Hub.Service/samples/IotHubClientSamples/CleanupHelper.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Azure.Iot.Hub.Service.Models;
+
+namespace Azure.Iot.Hub.Service.Samples
+{
+    /// <summary>
+    /// Helper class to help cleanup resources created by the sample runs.
+    /// </summary>
+    public static class CleanupHelper
+    {
+        public async static Task DeleteAllDevicesInHubAsync(IotHubServiceClient hubClient)
+        {
+            Console.WriteLine($"\nTrying to clean up all devices\n");
+
+            try
+            {
+                AsyncPageable<TwinData> asyncPageableResponse = hubClient.Devices.GetTwinsAsync();
+                List<TwinData> deviceTwins = new List<TwinData>();
+                await foreach (TwinData twin in asyncPageableResponse)
+                {
+                    deviceTwins.Add(twin);
+                }
+
+                foreach (TwinData twin in deviceTwins)
```

Comment:
This logic assumes all devices have a twin. Let me know if that is incorrect.

Reply:
Currently i'm doing 
1) List all twins
2) Get device Ids from there
3) Get every device identity using the id
4) Delete every device identity

Reply:
If there is not a better way to do this, we should discuss how we can expose List all device identities. Customers might have a similar scenario too.

Reply:
I don't see a way around Step 1, 2 and 3 ; but instead of calling Step 4 in a for-each loop, we could also use the bulk API.
I remember people being unhappy about this in the track 1 client, so we could probably expose an API that combines 1, 2 and 3 and returns the result?

Reply:
We should be careful in adding functionality that extends beyond the REST API. I'm not saying we don't, but let's treat that decision with care and thoroughly discuss it.